src/Analyzers/Core/Analyzers/Helpers/DiagnosticHelper.cs

```diff
@@ -206,55 +290,6 @@ static string EncodeIndices(IEnumerable<int> indices, int additionalLocationsLen
             }
         }
 
-        /// <summary>
-        /// Creates a <see cref="Diagnostic"/> instance.
-        /// </summary>
-        /// <param name="descriptor">A <see cref="DiagnosticDescriptor"/> describing the diagnostic.</param>
-        /// <param name="location">An optional primary location of the diagnostic. If null, <see cref="Location"/> will return <see cref="Location.None"/>.</param>
-        /// <param name="effectiveSeverity">Effective severity of the diagnostic.</param>
-        /// <param name="additionalLocations">
-        /// An optional set of additional locations related to the diagnostic.
-        /// Typically, these are locations of other items referenced in the message.
-        /// If null, <see cref="Diagnostic.AdditionalLocations"/> will return an empty list.
-        /// </param>
-        /// <param name="properties">
-        /// An optional set of name-value pairs by means of which the analyzer that creates the diagnostic
-        /// can convey more detailed information to the fixer. If null, <see cref="Diagnostic.Properties"/> will return
-        /// <see cref="ImmutableDictionary{TKey, TValue}.Empty"/>.
-        /// </param>
-        /// <param name="message">Localizable message for the diagnostic.</param>
-        /// <returns>The <see cref="Diagnostic"/> instance.</returns>
-        public static Diagnostic CreateWithMessage(
```

Comment:
I am still unsure if removing this overload and `LocalizableStringWithArguments` is required for this PR. Can you clarify why it is being done here?

Reply:
`LocalizableStringWithArguments` was originally introduced to support `option = value:none`. Since this PR introduces a new way to support none, the type is no longer needed.

This has also caught extra period in analyzer messages.